gulpfile.js

```diff
@@ -63,9 +63,13 @@ var GH_PAGES_DIR = BUILD_DIR + "gh-pages/";
 var SRC_DIR = "src/";
 var LIB_DIR = BUILD_DIR + "lib/";
 var DIST_DIR = BUILD_DIR + "dist/";
+var TYPES_BUILD_DIR = BUILD_DIR + "types/";
 var COMMON_WEB_FILES = ["web/images/*.{png,svg,gif,cur}", "web/debugger.js"];
 var MOZCENTRAL_DIFF_FILE = "mozcentral.diff";
 
+// This is either `tsc` or `jsdoc`
+var TYPES_INT = "tsc";
+
```

Comment:
Why do you need two *different* modes though, since I believe that it was mentioned (in the previous PRs) that the JSDoc solution wasn't "powerful"/"useful" enough.

Can the implementation be simplified to just use *one* mode, since having something that'd require *manual* editing of the gulpfile to toggle will basically render it unused any way?

Reply:
of course - this is still to test the difference. I did not see the reference to JSDoc not being "powerful" enough. There will only be one...

Reply:
removed